src/RA_AchievementSet.cpp

```diff
@@ -431,38 +364,39 @@ BOOL AchievementSet::LoadFromFile(ra::GameID nGameID)
     const std::string sFilename = GetAchievementSetFilename(nGameID);
 
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pFile = nullptr;
-    errno_t nErr = fopen_s(&pFile, sFilename.c_str(), "r");
-    if (pFile != nullptr)
+
+    if (std::ifstream ifile{ sFilename }; !ifile.is_open())
+    {
+        //	Cannot open file
+        RA_LOG("Cannot open file %s\n", sFilename.c_str());
+        RA_LOG("Error %s\n", NativeStr(ra::GetLastErrorMsg()).c_str());
```

Comment:
not sure where my previous comment went, but ifsteam sets errno, so calling GetLastError won't return the expected message.

Reply:
also RA_LOG expects MBCS strings. `NativeStr` should be replaced with `Narrow`

Reply:
Well, sometimes it returns "Cannot create a file that already exists", or "The file is in use by another process". So it seems legit to me.

Reply:
That's leftover from somewhere else:
```
    RA_LOG("Non-Error %s\n", NativeStr(ra::GetLastErrorMsg()).c_str());

    if (std::ifstream ifile{ sFilename }; !ifile.is_open())
    {
        //	Cannot open file
        RA_LOG("Cannot open file %s\n", sFilename.c_str());
        RA_LOG("Error %s\n", NativeStr(ra::GetLastErrorMsg()).c_str());
        return FALSE;
    }
```
```
Non-Error Cannot create a file when that file already exists.
Cannot open file .\RACache\Data\1466-User.txt
Error Cannot create a file when that file already exists.
```